src/Equations/Fluid/IonChargedAdvectionTerm.cpp

```diff
@@ -0,0 +1,100 @@
+#include "DREAM/Equations/Fluid/IonChargedAdvectionTerm.hpp"
+#include "FVM/Grid/Grid.hpp"
+
+#include <iostream>
+
+using namespace DREAM;
+
+IonChargedAdvectionTerm::IonChargedAdvectionTerm(FVM::Grid *grid, IonHandler *ions, const len_t iIon, 
+    bool allocCoefficients, FVM::AdvectionInterpolationCoefficient::adv_interpolation intp,
+    OptionConstants::adv_jacobian_mode jac_mode, len_t id, real_t damping_factor) : IonChargedAdvectionDiffusionTerm<FVM::AdvectionTerm>(grid, ions, iIon, allocCoefficients){
+	    std::cout<<"check charged 2 \n";
+	    AllocateInterpolationCoefficients();
+	    SetAdvectionInterpolationMethod(intp, jac_mode, id, damping_factor);
+	}
+	
+IonChargedAdvectionTerm::~IonChargedAdvectionTerm(){}
+
+void IonChargedAdvectionTerm::AllocateInterpolationCoefficients(){
+    if (!this->interpolationCoefficientsShared)
+        DeallocateInterpolationCoefficients();
+        
+    const len_t nr = this->grid->GetNr();
+
+    len_t idx = this->ions->GetIndex(iIon, 0);
+    for(len_t Z0 = 1; Z0<=Zion; Z0++,idx++)
+        this->deltars.push_back(new FVM::AdvectionInterpolationCoefficient(grid, FVM::FLUXGRIDTYPE_RADIAL, FVM::AdvectionInterpolationCoefficient::AD_BC_MIRRORED, FVM::AdvectionInterpolationCoefficient::AD_BC_DIRICHLET, idx*nr));
+    
+    this->interpolationCoefficientsShared = false;
+}
+
+void IonChargedAdvectionTerm::DeallocateInterpolationCoefficients(){
+    for (auto it = deltars.begin(); it != deltars.end(); it++)
```

Comment:
Just FYI C++ also has a more compact syntax for this:
```
for (auto dt : deltars)
```
In this case, ``dt`` will not be an iterator but a reference to the element directly (i.e. what you get when you dereference ``it`` now, ``*it``)